torch/_guards.py

```diff
@@ -748,6 +749,12 @@ def __init__(self, fake_mode):
         self.fakify_first_call = False
         self.hop_dispatch_set_cache = HopDispatchSetCache()
 
+        # To reuse invoke subgraph infrastructure, other hops that function similarly
+        # as invoke_subgraph get traced as invoke_subgraph with an identifier that we store here
+        # Then, after autograd tracing, we map them back to the correct hop
+        self.invoke_subgraph_mapping_counter = itertools.count(0)
+        self.invoke_subgraph_mapping: Dict[str, Any] = {}
+
```

Comment:
This is a bit hacky. From a code organization perspective:
- It's fine if we want InvokeSubgraph and InvokeQuant to inherit from some base class to share implementation details
- InvokeQuant probably shouldn't have an identifier. InvokeSubgraph should have an identifier. There doesn't seem to be a reason for InvokeQuant to take in an identifier